src/contracts/core/AVSDirectory.sol

```diff
@@ -581,42 +586,40 @@ contract AVSDirectory is
         }
     }
 
-    /**
-     * @notice Completes queued deallocations for a single strategy and multiple operator sets
-     * @param operator address to complete deallocations for
-     * @param strategy the strategy to complete deallocations for
-     * @param operatorSet the operator set to complete deallocations for the given strategy
-     */
-    function _completeDeallocation(
+    /// @dev read through pending free magnitudes and add to freeMagnitude if completableTimestamp is >= block timestamp
+    /// In additiona to updating freeMagnitude, updates next starting index to read from for pending free magnitudes after completing
+    function _updateFreeMagnitude(
         address operator,
-        IStrategy strategy,
-        OperatorSet calldata operatorSet
-    ) internal returns (uint64 freeMagnitudeToAdd) {
-        QueuedDeallocation[] memory queuedDeallocation =
-            _queuedDeallocations[operator][strategy][operatorSet.avs][operatorSet.operatorSetId];
-        // read last completed deallocation index in the queuedDeallocations array for the (op, opSet, Strategy) tuple
-        uint256 i = _nextDeallocationIndex[operator][strategy][operatorSet.avs][operatorSet.operatorSetId];
-
-        for (; i < queuedDeallocation.length; ++i) {
-            // queued deallocations ordered by timestamp, if completableTimestamp is greater than completeUntilTimestamp, break
-            if (queuedDeallocation[i].completableTimestamp > uint32(block.timestamp)) {
+        IStrategy strategy
+    ) internal {
+        uint256 nextIndex = _nextPendingFreeMagnitudeIndex[operator][strategy];
+        uint256 pendingFreeMagnitudeLength = _pendingFreeMagnitude[operator][strategy].length;
+        while (nextIndex < pendingFreeMagnitudeLength) {
+            PendingFreeMagnitude memory pendingFreeMagnitude = _pendingFreeMagnitude[operator][strategy][nextIndex];
+            // pendingFreeMagnitude is ordered by completableTimestamp. If we reach one that is not completable yet, then break
+            // loop until completableTimestamp is < block.timestamp
+            if (pendingFreeMagnitude.completableTimestamp < uint32(block.timestamp)) {
                 break;
             }
 
-            freeMagnitudeToAdd += queuedDeallocation[i].magnitudeDiff;
+            // pending free magnitude can be added to freeMagnitude
+            freeMagnitude[operator][strategy] += pendingFreeMagnitude.magnitudeDiff;
+            ++nextIndex;
         }
-        _nextDeallocationIndex[operator][strategy][operatorSet.avs][operatorSet.operatorSetId] = i;
-        return freeMagnitudeToAdd;
+        // update next pending free magnitude index to start from after adding all completable magnitudes
+        _nextPendingFreeMagnitudeIndex[operator][strategy] = nextIndex;
     }
 
     /// @dev Check for max number of pending deallocations, ensuring <= MAX_PENDING_UPDATES
```

Comment:
rename from pending dealllocations

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/680/commits/d33bceec29869d0ee995a73cefc5c98e08d72c27
Changed to `_checkQueuedDeallocations`, since we have pendingFreeMagnitude now